tests/publiccloud/registercloudguest.pm

```diff
@@ -0,0 +1,62 @@
+# SUSE's openQA tests
+#
+# Copyright 2022 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Package: cloud-regionsrv-client
+# Summary: Test system (re)registration
+# https://github.com/SUSE-Enceladus/cloud-regionsrv-client/blob/master/integration_test-process.txt
+#
+# Maintainer: <qa-c@suse.de>
+
+use Mojo::Base 'publiccloud::basetest';
+use version_utils;
+use registration;
+use warnings;
+use testapi;
+use strict;
+use utils;
+use publiccloud::utils;
+
+sub run {
+    my ($self, $args) = @_;
+
+    # Preserve args for post_fail_hook
+    $self->{instance} = $args->{my_instance};
+
+    # Create $instance to make the code easier to read
+    my $instance = $args->{my_instance};
+    my $regcode = get_required_var('SCC_REGCODE') if (is_byos());
+    my $regcode_param = ($regcode) ? "-r $regcode" : '';
+
+    select_host_console();    # select console on the host, not the PC instance
+
+    # Test re-registration. Assuming the system has been registered before
+    $instance->run_ssh_command(cmd => '! sudo SUSEConnect -d', quiet => 0) if (is_byos());
+    $instance->run_ssh_command(cmd => 'sudo registercloudguest --clean', quiet => 0);
```

Comment:
calling `registercloudguest --clean` just after `SUSEConnect -d` was called , I am not sure that this is good idea . Why you would do de-register two time in a row ?

Reply:
I test that `SUSEConnect -d` fails. See the [documentation](https://github.com/SUSE-Enceladus/cloud-regionsrv-client/blob/master/integration_test-process.txt):
> - SUSEConnect -d
>   + produce error message